homeassistant/components/sonos/media_player.py

```diff
@@ -638,16 +639,27 @@ def _play_media(
                 soco.add_to_queue(playlist, timeout=LONG_SERVICE_TIMEOUT)
                 soco.play_from_queue(0)
         elif media_type in PLAYABLE_MEDIA_TYPES:
-            item = media_browser.get_media(self.media.library, media_id, media_type)
-
-            if not item:
-                _LOGGER.error('Could not find "%s" in the library', media_id)
-                return
-
-            soco.play_uri(item.get_uri())
+            self._play_media_playable_media_types(
+                media_type=media_type, media_id=media_id
+            )
```

Comment:
Inlining the changes made the function too complex according to Ruff.  Moved the section out into a new method.